modules/tinymce/src/themes/silver/test/ts/browser/editor/header/PromotionTest.ts

```diff
@@ -0,0 +1,73 @@
+import { ApproxStructure, Assertions, UiFinder } from '@ephox/agar';
+import { context, describe, it } from '@ephox/bedrock-client';
+import { SugarElement } from '@ephox/sugar';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.themes.silver.editor.header.PromotionTest', () => {
+  const promotionClass = 'tox-promotion';
+
+  context('promotion turned off', () => {
+    const hook = TinyHooks.bddSetup<Editor>({
+      base_url: '/project/tinymce/js/tinymce',
+    }, []);
+
+    it('TINY-8840: promotion should not be displayed', () => {
+      const editor = hook.editor();
+
+      UiFinder.notExists(SugarElement.fromDom(editor.getContainer()), '.' + promotionClass);
+    });
+  });
+
+  context('promotion turned on', () => {
+    const hook = TinyHooks.bddSetup<Editor>({
+      base_url: '/project/tinymce/js/tinymce',
+      promotion: true
+    }, []);
+
+    it('TINY-8840: promotion should not be displayed', () => {
+      const editor = hook.editor();
+
+      const editorContainer = SugarElement.fromDom(editor.getContainer());
+      UiFinder.exists(editorContainer, '.' + promotionClass);
+      Assertions.assertStructure(
+        'Editor structure',
+        ApproxStructure.build((s, str, arr) => s.element('div', {
+          classes: [ arr.has('tox-tinymce') ],
+          children: [
+            s.element('div', {
+              classes: [ arr.has('tox-editor-container') ],
+              children: [
+                s.element('div', {
+                  classes: [ arr.has('tox-editor-header') ],
+                  children: [
+                    s.element('div', {
+                      classes: [ arr.has('tox-promotion') ],
+                      children: [
+                        s.element('a', {
+                          classes: [ arr.has('tox-promotion-link') ],
+                          attrs: {
+                            rel: str.is('noopener'),
+                            target: str.is('_blank'),
+                            href: str.startsWith('http')
+                          }
+                        }),
+                      ]
+                    }),
+                    s.element('div', {}),
```

Comment:
This really should be checking the next element is the menubar, otherwise it could be something else unexpected.

Also we could simplify this/remove some of the unclear `s.element('div', {})` calls if instead we only do the assertion on the `tox-editor-header`. For example:
```ts
const editorContainer = TinyDom.container(editor);
const header = SelectorFind.descendant(editorContainer, '.tox-editor-header');
Assertions.assertStructure(
  'Editor header structure',
  ApproxStructure.build((s, str, arr) => s.element('div', {
    ...
  })),
  header
);
```
We also could consider using `s.theRest()` or `s.oneOrMore()` if we don't care about the contents of the elements after.